lib/rdf2marc/queries/contributions.rb

```diff
@@ -52,11 +52,15 @@ def resource_contributors_of_type(contributor_node, agent_types)
         return if contributor_terms.blank?
 
         contributor_terms.map do |contrib_term|
-          contributor_type = path_first([[RDF.type]], subject_term: contrib_term)
-          agent_types.flatten.include?(contributor_type) ? contrib_term : nil
+          contributor_types = path_all([[RDF.type]], subject_term: contrib_term)
+          overlap?(agent_types.flatten, contributor_types) ? contrib_term : nil
         end.compact
       end
 
+      def overlap?(array1, array2)
```

Comment:
Overlap is ambiguous as to whether it is a partial or complete. How about calling this `intersect?` instead. Then you could use `array1.intersection(array2).present?`

Reply:
Definitely. Fixed.